### PR TITLE
Fixes #30966 - make host substatus clickable

### PR DESCRIPTION
**Pull request by yifatmakias**
Made host substatus on the hosts/show page clickable.

### CODE REVIEW

**Review by tbrisker — CHANGES_REQUESTED** [review]
How is this different to GH-8043? 
Also, `arf_reports_path` is defined by the openscap plugin, meaning this won't work when it isn't installed. It will also link from the global status to the arf reports path, while the global status could come from multiple other sources.

**Comment by yifatmakias** [comment]
@tbrisker Regarding this: How is this different to GH-8043? - I closed the other PR.
Regarding the other issue, I see now that it is problematic. I will think about how to fix it.

**Comment by yifatmakias** [comment]
[test katello]

File: app/helpers/host_description_helper.rb [hunk 1]
```diff
@@ -35,8 +35,9 @@ def base_status_overview_fields(host)
       {
         :field => [
           _("Status"),
-          content_tag(:span, ''.html_safe, :class => host_global_status_icon_class(global_status.status)) +
+          link_to(content_tag(:span, ''.html_safe, :class => host_global_status_icon_class(global_status.status)) +
             content_tag(:span, _(global_status.to_label), :class => host_global_status_class(global_status.status)),
+            arf_reports_path(:search => "host = #{host.id}")),
```
Comment (timogoebel) [inline app/helpers/host_description_helper.rb]: Does core know this?

**Comment by xprazak2** [comment]
2 things that I see here:

* The method you are modifying is for global status, you might need to change the one that is for [substatuses](https://github.com/theforeman/foreman/blob/develop/app/helpers/host_description_helper.rb#L49). It goes though each host status and generates an entry in table for it:

statuses

* To avoid using plugin path helpers in core, you need to abstract it somehow. Slightly extending status might work here:

```ruby
module HostStatus
  class Status < ApplicationRecord
    def status_link
    end
  end
end
```

Then you can provide the link in foreman_openscap:

```ruby
module ForemanOpenscap
  class ComplianceStatus < ::HostStatus::Status
    def status_link
      '/path/to/compliance/status'
    end
  end
end
```

Back in core template, the helper would take a look at each status and generate link if the status has path for it.

**Comment by yifatmakias** [comment]
related to this PR:
https://github.com/theforeman/foreman_openscap/pull/452

**Comment by yifatmakias** [comment]
[test katello]

**Comment by yifatmakias** [comment]
[test katello]

**Comment by tbrisker** [comment]
Katello tests need rebasing on latest develop to pass

**Comment by xprazak2** [comment]
Nice, would it be possible to add an underline on link hover, same as the 'clear' link has?

**Comment by xprazak2** [comment]
Could you also change the commit message as this makes it possible for any host substatus to be a link?

**Comment by yifatmakias** [comment]
@xprazak2 I added an underline on link hover and changed the commit message. Let me know if it is ok now :)

File: app/helpers/hosts_helper.rb [hunk 2]
```diff
@@ -135,6 +135,19 @@ def host_global_status_class(status)
     end
   end
 
+  def host_global_status_link_class(status)
+    case status
```
Comment (xprazak2) [inline app/helpers/hosts_helper.rb]: I think you only want to add class for underline `if status.status_link`

**Comment by xprazak2** [comment]
Looks good, one minor thing is that the underline shows also for items that are not links.

File: app/helpers/hosts_helper.rb [hunk 3]
```diff
@@ -135,6 +135,23 @@ def host_global_status_class(status)
     end
   end
 
+  def host_global_status_link_class(status)
+    if status.status_link
```
Comment (xprazak2) [inline app/helpers/hosts_helper.rb:139]: Undefined method `status_link` for integer:

status-link

Caused by giving `status.to_global` as an arg when the method is called in host_description_helper.rb#56

Comment (yifatmakias) [reply app/helpers/hosts_helper.rb:139]: I fixed this bug it should not happen again.

**Review by xprazak2 — APPROVED** [review]
Works nicely, any additional concerns @tbrisker?

**Comment by ares** [comment]
this would be good to have in 2.3, so plugins can start using it (not sure about risks though, I haven't looked at the code)

**Review by tbrisker — APPROVED** [review]
LGTM, thanks @yifatmakias !

**Pull request merged**